src/schoology-extractor/tests/api/test_paginated_result.py

```diff
@@ -0,0 +1,121 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import pytest
+
+from schoology_extractor.api.paginated_result import PaginatedResult
+from schoology_extractor.api.request_client_base import RequestClientBase
+
+fake_key = 'TEST_KEY'
+fake_secret = 'TEST_SECRET'
+fake_endpoint_url = 'FAKE_URL'
+default_url = 'https://api.schoology.com/v1/'
+fake_resource_name = 'resource_name'
+fake_api_response = {
+    'total': 2,
+    'links': {
+        'next': 'url'
+    }
+}
+
+
+@pytest.fixture
+def default_request_client(mocker):
+    request_client = RequestClientBase(fake_key, fake_secret)
+    request_client.get = mocker.MagicMock()
+    return request_client
+
+
+@pytest.fixture
+def default_paginated_result(default_request_client, mocker):
+    return PaginatedResult(
+        request_client_base=default_request_client,
+        page_size=20,
+        api_response=fake_api_response,
+        resource_name=fake_resource_name,
+        requested_url=fake_endpoint_url)
+
+
+class TestPaginatedResult:
+    class Test_when_constructing:
+        def test_given_no_parameters_shoud_throw_assertException(self):
+
+            # Assert
+            with pytest.raises(AssertionError):
+                PaginatedResult(None, None, None, None, None)
+
+        def test_given_wrong_types_in_parameters_shoud_throw_assertException(self):
+
+            # Assert
+            with pytest.raises(AssertionError):
+                PaginatedResult(0, 1, 'None', [], '')
```

Comment:
@stephenfuqua @lalrg  I'm not seeing the cost/benefit of these wrong type tests if we are going to use type annotations, which I think we should.  For example, here and on line 47 the Pylance extension highlights them as errors in VS Code.

If you want to keep them around you'll need use the standard `# type: ignore` to suppress the errors.

Reply:
I have yet to see any advantage to type annotations other than to the reader. Even if Pylance flags something in the IDE - which it isn't doing for me (I'll do some reading to try to understand why) - will that prevent something strange from happening at runtime?

Avoiding the equivalent of null reference errors, and avoiding accidental empty strings, is important to me for preventing surprise bugs, the kinds of bugs that I have seen and had difficulty debugging in the past. So I would rather throw an error proactively then let them bubble up from deeper in the code, so that I can have better control over the messaging and understanding what is actually occurring.

I also have no idea what this comment about `# type: ignore` is referring to.

Reply:
Splitting my comments up here:
1) My IDE experience with type annotations has been fantastic.  It's equivalent to the difference between writing plain Javascript and writing TypeScript.  I strongly advocate for their use and am happy to help with any set up issues.
2) No argument with the usage of assert in the code base.
3) However, I see little value in writing tests that only highlight the same issues that a type checker will highlight.
4) If we want to write those tests anyway, put a `# type: ignore` comment on the end of the offending line.  In this example, line 53.